src/components/connect.js

```diff
@@ -103,11 +103,15 @@ export default function connect(mapPropsToRequestsToProps, options = {}) {
   }
 
   function handleResponse(response) {
-    const json = response.json() // TODO: support other response types
-    if (response.status >= 200 && response.status < 300) { // TODO: support custom acceptable statuses
-      return json
-    } else {
-      return json.then(cause => Promise.reject(newError(cause)))
+    // TODO: support other response types
+    if (/^application\/json/.test(response.headers.get('content-type'))) {
```

Comment:
I'm unsure how compatible this is with all derivative forms of the JSON MIME types (things like JSON API, etc.).

Also not sure if ancillary information like `;charset=UTF-8` is always guaranteed to come *after* the actual MIME type. Would appreciate some insight here.